src/session/content/chat_history.rs

```diff
@@ -163,6 +169,51 @@ impl ChatHistory {
                 _ => self.action_set_enabled("chat-history.view-info", false),
             }
 
+            let title_expression =
+                gtk::PropertyExpression::new(Chat::static_type(), gtk::NONE_EXPRESSION, "title");
+
+            match interlocutor_id(chat) {
+                Some(interlocutor_id) => {
+                    let session_expression = gtk::PropertyExpression::new(
```

Comment:
Creating expressions like this creates the problem that we had with sidebar-row: expressions won't unwatch themself automatically when new ones are created after another call to `set_chat()`.

We have three ways to fix this:
1. Do what I did in sidebar-row (basically storing all the bindings and unwatching before creating new ones);
2. Make the expressions autoupdate themselves (not always possible). To do that we have to create the expressions just a single time (ideally at `constructed()`) and binding to the internal property for making them updating themselves (in our case the `chat` property);
3. Use manual code instead of expressions. That ofc means that we won't get updates like expressions do.

For this case I suggest using 1. 2 could also work, but I think it can become too complicated (we'd be forced to have a bind of `me_expression` even with non-private chats).

Reply:
As suggested, I applied the first solution.